diff --git a/Makefile b/Makefile
index 460d44e..4e214b5 100644
--- a/Makefile
+++ b/Makefile
@@ -4,13 +4,13 @@ OBJECTS_O = onvif_simple_server.o device_service.o media_service.o ptz_service.o
 OBJECTS_N = onvif_notify_server.o conf.o utils.o log.o ezxml_wrapper.o ezxml/ezxml.o
 OBJECTS_W = wsd_simple_server.o utils.o log.o ezxml_wrapper.o ezxml/ezxml.o
 ifdef HAVE_MBEDTLS
-INCLUDE = -DHAVE_MBEDTLS -Iextras/mbedtls/include -ffunction-sections -fdata-sections -lrt
-LIBS_O = -Wl,--gc-sections extras/mbedtls/library/libmbedcrypto.a -lpthread -lrt
-LIBS_N = -Wl,--gc-sections extras/mbedtls/library/libmbedcrypto.a -lpthread -lrt
+INCLUDE = -DHAVE_MBEDTLS -I$(STAGING_DIR)/usr/include -ffunction-sections -fdata-sections -lrt
+LIBS_O = -Wl,--gc-sections mbedtls/library/libmbedcrypto.a -lpthread -lrt
+LIBS_N = -Wl,--gc-sections mbedtls/library/libmbedcrypto.a -lpthread -lrt
 else
-INCLUDE = -Iextras/libtomcrypt/src/headers -ffunction-sections -fdata-sections -lrt
-LIBS_O = -Wl,--gc-sections extras/libtomcrypt/libtomcrypt.a -lpthread -lrt
-LIBS_N = -Wl,--gc-sections extras/libtomcrypt/libtomcrypt.a -lpthread -lrt
+INCLUDE = -I$(STAGING_DIR)/usr/include  -ffunction-sections -fdata-sections -lrt
+LIBS_O = -Wl,--gc-sections $(STAGING_DIR)/usr/lib/libtomcrypt.a -lpthread -lrt
+LIBS_N = -Wl,--gc-sections $(STAGING_DIR)/usr/lib/libtomcrypt.a -lpthread -lrt
 endif
 LIBS_W = -Wl,--gc-sections
